src/diff/nodeChanges.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as AmfGraphTypes from "./amfGraphTypes";
+import { CategorizedChange } from "./categorizedChange";
+import { RuleCategory } from "./ruleSet";
+
+/**
+ * Class to hold differences of a JSON node
+ */
+export class NodeChanges {
+  public added?: { [key: string]: unknown } | AmfGraphTypes.Node;
+  public removed?: { [key: string]: unknown } | AmfGraphTypes.Node;
+  //categorized changes of the node
+  public categorizedChanges?: CategorizedChange[];
```

Comment:
I was debating on that since they are really optional

Reply:
Is there any meaningful difference between the property being unset and the property being set to an empty array?

Reply:
Not really. I don't think we want to have that difference.

It depends on how we want our interface to look for the user - undefined or empty array. The general approach seems to be undefined (based on the other code that I have seen). Since we have helper functions I think its fine to have as an empty array